### ISSUE TITLE
What is the expected behavior for invalid CSS?

### ISSUE THREAD

**Issue by alexweissman** [jakubpawlowicz/clean-css#754]
I inadvertently fed some invalid/malformed CSS into clean-css (using the now-deprecated `gulp-minify-css` package):

```
body .container {
  color: #EEEEEE;
}

h1{
  color: #000;
}

.navbar-default .navbar-nav > .active > a:hover, {
    color: black;
}
```

Notice the comma after `a:hover` before the start of the declaration block.  I believe this is invalid syntax, although most browsers seem to quietly ignore it and proceed with the declaration anyway.

However, after I run it through clean-css, I get some unexpected output:
`body .container{color:#EEE},.navbar-default .navbar-nav>.active>a:hover,h1{color:#000}`

Notice the comma after `body .container{color:#EEE}`.  According to my [question on SO](http://stackoverflow.com/questions/36408522/in-css3-what-does-a-comma-between-declaration-blocks-mean), it is not valid to have a comma after a declaration block.  What's more, using this minified CSS my browsers (FF and Chrome) seem to ignore the entire declaration after the comma, leading to incorrect styling.

Is this the intended behavior?  Should I add a separate CSS linter to my build task ahead of clean-css?

**Comment by jakubpawlowicz** [comment]
This is a very fine point @alexweissman and the whole declaration should be dropped instead, unless there is a browser which drops the comma and applies it fine. Could you also test in IE?

**Comment by alexweissman** [comment]
It would appear that the behavior is the same in IE.  The erroneous `,` is ignored in the unminified CSS and the styles are applied as expected, but in the minified CSS, the entire declaration block is dropped.

**Comment by jakubpawlowicz** [comment]
Thanks for checking. Then IE treats it differently than Chrome and Firefox where the whole rule is not applied. We'd need to keep it as it is and don't merge it with any other rules.

**Comment by jakubpawlowicz** [comment]
Fixed in 3.4.12.
